### PR TITLE
Misc enhancements

### PR DESCRIPTION
**Pull request by bowmanb**
Includes using dictionary instead of data attributes and adding some more hooks from WebViewClient I needed.

### CODE REVIEW

File: turbolinks/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ android {
     buildToolsVersion "23.0.2"
 
     defaultConfig {
-        minSdkVersion 19
+        minSdkVersion 21
```
Comment (joemasilotti) [inline turbolinks/build.gradle]: What's the adoption rate for 21?

Comment (joemasilotti) [inline turbolinks/build.gradle]: Never mind, I just noticed the last commit.

**Comment by joemasilotti** [comment]
Looks good to me. Good to merge, @bowmanb?

File: turbolinks/build.gradle (hunk 1 above)
Comment (bowmanb) [inline turbolinks/build.gradle]: I meant to, and did, lower it back down to 19. About 1/3 of users are on 19 last I checked.

**Comment by bowmanb** [comment]
Yup.

**Pull request merged**